lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_playbook_method.rb

```diff
@@ -50,109 +23,13 @@ def execute
       running_in_state_machine? ? check_task_status(task_id) : wait_for_method(task_id)
     end
 
-    def running_in_state_machine?
-      @workspace.root['ae_state_started'].present?
-    end
-
-    def process_result
-      @aw.reload
-      @workspace.update_workspace(@aw.output) if @aw.output
-      reset
-    end
-
-    def reset
-      @aw.delete
-      self.class.cleanup(@workspace)
-    end
-
-    def wait_for_method(task_id)
-      task = MiqTask.wait_for_taskid(task_id)
-      raise MiqAeException::Error, task.message unless task.status == "Ok"
-      process_result
-    ensure
-      reset
-    end
-
-    def check_task_status(task_id)
-      task = MiqTask.find(task_id)
-      raise MiqAeException::Error, "Task id #{task_id} not found" unless task
-      if task.state == MiqTask::STATE_FINISHED
-        if task.status == MiqTask::STATUS_OK
-          @workspace.root['ae_result'] = 'ok'
-          process_result
-        else
-          @workspace.root['ae_result'] = 'error'
-          reset
-          raise MiqAeException::Error, task.message
-        end
-      else
-        mark_for_retry(task_id)
-      end
-    end
-
-    def mark_for_retry(task_id)
-      @workspace.root['ae_result'] = 'async_launch'
-      @workspace.root['ae_retry_interval'] = retry_interval
-      @workspace.persist_state_hash['automate_workspace_guid'] = @aw.guid
-      @workspace.persist_state_hash[method_key] = task_id
-      $miq_ae_logger.info("Setting State Machine Auto Retry Interval: #{@workspace.root['ae_retry_interval']}")
-    end
-
-    def ttl
-      @aem.options[:execution_ttl].to_i.minutes
-    end
-
-    def max_retries
-      @workspace.root['ae_state_max_retries'].to_i
-    end
-
-    def retry_interval
-      interval = ttl.positive? && max_retries.positive? ? ttl / max_retries : MIN_RETRY_INTERVAL
-      interval > MIN_RETRY_INTERVAL ? interval : MIN_RETRY_INTERVAL
-    end
-
-    def method_manageiq_env
-      {
-        'automate_workspace' => @aw.href_slug
-      }.merge(manageiq_env(@workspace.ae_user, @workspace.ae_user.current_group, miq_request_task))
-    end
-
-    def method_key
-      @method_key_value ||= "#{@aem.name}#{METHOD_KEY_SUFFIX}".gsub(/\s+/, "")
-    end
-
-    def serialize_workspace
-      {'objects'           => @workspace.hash_workspace,
-       'method_parameters' => MiqAeReference.encode(@inputs),
-       'current'           => current_info,
-       'state_vars'        => MiqAeReference.encode(@workspace.persist_state_hash)}
-    end
-
-    def current_info
-      list = %w(namespace class instance message method)
-      list.each.with_object({}) { |m, hash| hash[m] = @workspace.send("current_#{m}".to_sym) }
-    end
-
-    def build_options_hash
+    def build_playbook_options_hash
+      build_options_hash
```

Comment:
@lfu can we call the method build_options_hash and call the parent classes build_options_hash and then add the extra stuff. 
```
def build_options_hash
   super.tap do |config_info|
      config_info[:hosts] = resolved_hosts
   end
end
```